### ISSUE TITLE
classDiagram Generics Map

### ISSUE THREAD

**Issue by vorburger** [mermaid-js/mermaid#4431]
**Is your feature request related to a problem? Please describe.**

On http://localhost:8000/concepts/core-arch/ I originally used this for [generics](https://mermaid.js.org/syntax/classDiagram.html#generic-types) in a Map:

``` mermaid
classDiagram
  class Entity{
    Timestamp ts
    Map~String, String~ links
  }
```

which currently fails (also here on GitHub) with:

```
caught RangeError: Maximum call stack size exceeded
    at String.replace (<anonymous>)
    at Uk (common.ts:164:36)
    at Uk (common.ts:167:12)
    at Uk (common.ts:167:12)
    at Uk (common.ts:167:12)
    at Uk (common.ts:167:12)
 ```

I gathered with a bit trial and error that this is due to the `,` comma in `Map~String, String~ ` (which I wrote like that coming from a Java-like `Map<String, String>` syntax).

I have worked around it by using a `|` instead of a `,` for now, so this works, but only if there are no spaces around it:

``` mermaid
classDiagram
  class Entity{
    Timestamp ts
    Map~String|String~ links
  }
```

But perhaps you would like to consider permitting a comma as well? And spaces? (Perhaps even some other separate characters?)

**Comment by turtton** [comment]
I also face the same issue.
It is a side note, but the same error occurs when you add spaces before and after the `|` symbol. I discovered this while expressing types similar to TypeScript.

```mermaid
classDiagram
class A {
~String | int~ value
}

**Comment by jakraft** [comment]
I hit into the same issue #4555. The only "true" workaround I know of is to avoid the tilde syntax entirely and use HTML-escaped angle brackets (`&lt;` and `&gt;`), however, this significantly reduces the readability of the diagram when viewing the plaintext. The better option, IMO, would be to replace the tilde syntax and just add support for angle brackets, which eliminates all ambiguities when parsing.

**Comment by jgreywolf** [comment]
This should be resolved with PR #4534
